ui/components/src/utils/memo.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 The Perses Authors
```

Comment:
Was there a circular dependency issue with `components` just using this from `core`? I think a dependency from components -> core is totally fine if we need to. Feels like `core` is always gonna be our "lowest level" package that others might depend on.

Reply:
It doesn't seem to be a circular dependency, but I'm seeing [these errors](https://github.com/perses/perses/runs/6181994115?check_suite_focus=true) in CI. I wasn't seeing them locally earlier, but I'll try again to reproduce it.

Reply:
Fixed in [this commit](https://github.com/perses/perses/pull/425/commits/ab2d991821f70e8c2d5be04b1b3ef1d325549520) so now components imports useDeepMemo from core